src/MacroExpander.js

```diff
@@ -25,18 +32,87 @@ MacroExpander.prototype.nextToken = function() {
         if (!(name.charAt(0) === "\\" && this.macros.hasOwnProperty(name))) {
             return topToken;
         }
+        var tok;
         var expansion = this.macros[name];
         if (typeof expansion === "string") {
+            var numArgs = 0;
+            if (expansion.indexOf("#") !== -1) {
+                var stripped = expansion.replace(/##/g, "");
```

Comment:
I think ## is supposed to expand to # in TeX (a way of quoting # that isn't part of an argument like #1).  Presumably this is to ignore those for detecting macro arguments.